app/models/katello/host/content_facet.rb

```diff
@@ -164,10 +164,12 @@ def update_repositories_by_paths(paths)
         prefixes = %w(/pulp/deb/ /pulp/repos/ /pulp/content/)
         relative_paths = []
 
-        # paths == ["/pulp/content/Default_Organization/Library/custom/Test_product/test2"]
+        # paths == ["/pulp/content/Default_Organization/Library/custom/Test_product/test2",
+        #           "/pulp/content/Org/Library/custom/Test_product/test2/%3Fcomp%3Dmain%26rel%3Dstable"]
         paths.each do |path|
           if (prefix = prefixes.find { |pre| path.start_with?(pre) })
-            relative_paths << path.gsub(prefix, '')
+            # strip prefix and structured_apt postfix before adding to relative_paths
+            relative_paths << path.sub(prefix, '').sub(/\/?%3F.*/, '')
```

Comment:
Would it be safer to parse the code (decode the "%3F" to "?" etc) and then URI parse it like:
```
=> "blub?comp=main"
irb(main):014:0> uri = URI.parse(a)
=> #<URI::Generic blub?comp=main>
irb(main):015:0> uri.path
=> "blub"
irb(main):016:0> query_params = URI.decode_www_form(uri.query).to_h
=> {"comp"=>"main"}
```

Reply:
We have made it robust for both the encoded and not encoded variants.